app/react/components/InsightsBox/InsightsBox.tsx

```diff
@@ -0,0 +1,68 @@
+import clsx from 'clsx';
+import { Lightbulb, X } from 'lucide-react';
+import { ReactNode, useMemo } from 'react';
+import sanitize from 'sanitize-html';
+import { useStore } from 'zustand';
+
+import { Button } from '@@/buttons';
+
+import { insightStore } from './insights-store';
+
+export type Props = {
+  header: string;
+  content: ReactNode;
+  setHtmlContent?: boolean;
+  insightCloseId?: string; // set if you want to be able to close the box and not show it again
+};
+
+export function InsightsBox({
+  header,
+  content,
+  setHtmlContent,
+  insightCloseId,
+}: Props) {
+  // allow to close the box and not show it again in local storage with zustand
+  const { insightIDsClosed, addInsightIDClosed } = useStore(insightStore);
+  const isInsightClosed = useMemo(() => {
+    if (insightCloseId) {
+      return insightIDsClosed.includes(insightCloseId);
+    }
+    return false;
+  }, [insightCloseId, insightIDsClosed]);
+
+  // allow angular views to set html messages for the insights box
+  const htmlContent = useMemo(() => {
```

Comment:
maybe instead of allowing angular views to pass HTML text (a functionality that we will need to remember to remove in the future), make a react component that uses this component for every place we need this in angular?

Reply:
I would prefer to keep this as is for now, if that's ok? When angular is gone, If I made react components for these sections, I would likely rework these when angular is gone

Reply:
that's your decision, but that's adding more overhead for components that we don't really need.

if you don't want to create a component for the whole tooltip, you can create a component just for the tooltip HTML:

*insights.tsx
```tsx
export function Insights() {
return <>
	<p>
                    From 2.18 on, the set-up of available GPUs for a Docker Standalone environment has been shifted from Add environment and Environment details to Host -> Setup, so as to align with other settings.
                  </p>
                  <p>
                    A toggle has been introduced for enabling/disabling management of GPU settings in the Portainer UI - to alleviate the performance impact of showing those settings.
                  </p>
                  <p>
                    The UI has been updated to clarify that GPU settings support is only for Docker Standalone (and not Docker Swarm, which was never supported in the UI).
                  </p>
</>
}
```

and just import it to your controller:

```ts
import {Insights } from './insights.tsx'

function controller() {
   this.insights = Insights 
}
```

and in the template:

```html
<insights content="$ctrl.insights"></insights>
```

wdyt? it will just reduce the changes we need to do once we migrate to react, and removing the props and the `useMemo` overhead